cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -177,16 +175,14 @@ public final ProgramController run(final Program program, final ProgramOptions o
                             new LocalizeResource(saveHConf(hConf, File.createTempFile("hConf", ".xml", tempDir))));
       localizeResources.put("cConf.xml",
                             new LocalizeResource(saveCConf(cConf, File.createTempFile("cConf", ".xml", tempDir))));
-      File programDir = DirUtils.createTempDir(tempDir);
-      final Program copiedProgram = copyProgramJar(program, tempDir, programDir);
```

Comment:
Just curious - why don't we need to copy the program jar anymore? Or why were we copying the jar before? Was it just to unpack  the jar?